### PR TITLE
General page layout

### PR DESCRIPTION
**Pull request by rfejzic1**
General page layout set up. Added a navbar, breadcrumbs and a footer. More styling needs to be done and the App.scss file needs to be better modularized.

### CODE REVIEW

File: auction/src/components/Common/Container.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import React from 'react'
+
+const Container = props => {
+    const { title, children } = props;
+    return (
+        <div className='container'>
+            <div className='container-header'><h2>{title}</h2></div>
```
Comment (mirzaAlispahic) [inline auction/src/components/Common/Container.js]: put h2 in new line

File: auction/src/components/Common/SecureRoute.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+            {...rest}
+            render={props =>
+                loggedIn ? 
+                <Component {...props} />
+                :
+                <Redirect
+                    to={{
+                        pathname: "/login",
+                        state: { referrer: props.location }
+                    }}
```
Comment (mirzaAlispahic) [inline auction/src/components/Common/SecureRoute.js:21]: move this inner object to variable or const

File: auction/src/components/Common/TopBar.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+                <Social/>
+                {
+                    loggedIn ?
+                    <span>
+                        <Link 
+                        to='#'
+                        onClick={e => {
+                            e.preventDefault();
+                            logout(dispatch);
+                        }}>Logout</Link>
```
Comment (mirzaAlispahic) [inline auction/src/components/Common/TopBar.js]: Move to function outside return statement.

File: auction/src/components/Controls/SelectField.js [hunk 4]
```diff
@@ -0,0 +1,15 @@
+import React from 'react'
+import classNames from 'classnames';
+
+const SelectField = props => {
+    const { name, id, options, fullWidth, onChange } = props;
+    const classes = classNames('select-field', {'full-width': fullWidth});
+
+    return (
+        <select onChange={onChange} className={classes} name={name} id={id}>
+            {options.map((option, index) => <option value={index}>{option}</option>)}
```
Comment (mirzaAlispahic) [inline auction/src/components/Controls/SelectField.js]: `{
    options.map((option, index) => {
       <option value={index}>{option}</option>)
    }
}` and move `<option value={index}>{option}</option>` to new line

File: auction/src/components/Controls/TextField.js [hunk 5]
```diff
@@ -0,0 +1,14 @@
+import React from 'react'
+import classNames from 'classnames';
+
+const TextField = props => {
+    const { placeholder, name, password, fullWidth, onChange, className } = props;
+    const type = password ? 'password' : 'text';
+    const classes = classNames('text-field', {'full-width': fullWidth}, className);
```
Comment (mirzaAlispahic) [inline auction/src/components/Controls/TextField.js]: new line for each class

File: auction/src/components/Login.js [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+            {userData.loggedIn ?
+            <Redirect to={redirectPath} />
+            :
+            <>
+            <Breadcrumbs current='login'/>
+            <Divider/>
+            <Container title='LOGIN'>
+                <FormGroup>
+                    <Label isFor='email' label='Email' />
+                    <TextField onChange={e => setEmail(e.target.value)} name='email' placeholder='e.g. johndoe@mail.com' fullWidth />
```
Comment (mirzaAlispahic) [inline auction/src/components/Login.js]: move to function above return statement.

File: auction/src/components/Login.js [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+                </FormGroup>
+                <FormGroup>
+                    <Label isFor='password' label='Password' />
+                    <TextField onChange={e => setPassword(e.target.value)} name='password' fullWidth password />
```
Comment (mirzaAlispahic) [inline auction/src/components/Login.js]: same comment as above.

File: auction/src/components/Login.js [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+                </FormGroup>
+                <FormGroup>
+                    <Button 
+                        onClick={() => login(dispatch, { email, password })}
```
Comment (mirzaAlispahic) [inline auction/src/components/Login.js]: same as above.

File: auction/src/components/Register.js [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+    const [password, setPassword] = useState('');
+
+    return (
+        <React.Fragment>
+            <Breadcrumbs current='register'/>
+            <Divider/>
+            <Container title='REGISTER'>
+                <FormGroup>
+                    <Label isFor='firstName' label='First Name' />
+                    <TextField onChange={e => setFirstName(e.target.value)} name='firstName' placeholder='e.g. John' fullWidth />
```
Comment (mirzaAlispahic) [inline auction/src/components/Register.js]: move above return statemenent.

File: auction/src/components/Register.js [hunk 10]
```diff
@@ -0,0 +1,63 @@
...
+                </FormGroup>
+
+                <FormGroup>
+                    <Label isFor='lastName' label='Last Name' />
+                    <TextField onChange={e => setLastName(e.target.value)} name='lastName' placeholder='e.g. Doe' fullWidth />
```
Comment (mirzaAlispahic) [inline auction/src/components/Register.js]: move above return statemenent.

File: auction/src/components/Register.js [hunk 11]
```diff
@@ -0,0 +1,63 @@
...
+                </FormGroup>
+
+                <FormGroup>
+                    <Label isFor='email' label='Email' />
+                    <TextField onChange={e => setEmail(e.target.value)} name='email' placeholder='e.g. johndoe@mail.com' fullWidth />
```
Comment (mirzaAlispahic) [inline auction/src/components/Register.js]: move above return statemenent.

File: auction/src/components/Register.js [hunk 12]
```diff
@@ -0,0 +1,63 @@
...
+                </FormGroup>
+
+                <FormGroup>
+                    <Label isFor='password' label='Password' />
+                    <TextField onChange={e => setPassword(e.target.value)} name='password' fullWidth password />
```
Comment (mirzaAlispahic) [inline auction/src/components/Register.js]: move above return statemenent.

File: auction/src/components/Register.js [hunk 13]
```diff
@@ -0,0 +1,63 @@
...
+                </FormGroup>
+
+                <FormGroup>
+                    <Button 
+                        onClick={() => console.log(`First name: ${firstName}\nLast name: ${lastName}\nemail: ${email}\npassword: ${password}`)}
```
Comment (mirzaAlispahic) [inline auction/src/components/Register.js]: remove console.log

File: auction/src/services/SessionService.js [hunk 14]
```diff
@@ -0,0 +1,31 @@
+import axios from 'axios';
+import Cookies from 'js-cookie';
+
+function setAuthorizationHeaderDefault(token) {
+    axios.defaults.headers.common['Authorization'] = `Bearer ${token}`;
+}
+
+export const login = (dispatch, loginData) => {
+    axios.post('http://localhost:9000/login', loginData)
```
Comment (mirzaAlispahic) [inline auction/src/services/SessionService.js]: move `http://localhost:9000` to constant which you will export from another js file. Use import to get that constant.

**Review by mirzaAlispahic — CHANGES_REQUESTED** [review]

File: auction/src/components/Common/Container.js (hunk 1 above)
Comment (rfejzic1) [reply auction/src/components/Common/Container.js]: Done

File: auction/src/components/Common/TopBar.js (hunk 3 above)
Comment (rfejzic1) [reply auction/src/components/Common/TopBar.js]: Done

File: auction/src/components/Controls/TextField.js (hunk 5 above)
Comment (rfejzic1) [reply auction/src/components/Controls/TextField.js]: Done

File: auction/src/components/Login.js (hunk 6 above)
Comment (rfejzic1) [reply auction/src/components/Login.js]: Done

File: auction/src/components/Login.js (hunk 8 above)
Comment (rfejzic1) [reply auction/src/components/Login.js]: Done

File: auction/src/services/SessionService.js (hunk 14 above)
Comment (rfejzic1) [reply auction/src/services/SessionService.js]: Done

File: auction/src/components/Register.js [hunk 15]
```diff
@@ -0,0 +1,84 @@
...
+                </FormGroup>
+
+                <FormGroup flex around>
+                    <span>
+                        Already have an account? <Link to='/login'>Login</Link>
+                    </span>
+                </FormGroup>
+            </Container>
+            <Divider/>
+        </>
```
Comment (mirzaAlispahic) [inline auction/src/components/Register.js]: Indentation.

File: auction/src/services/SessionService.js [hunk 16]
```diff
@@ -0,0 +1,46 @@
...
+
+export const setAPIBaseURL = domain => {
+    axios.defaults.baseURL = domain;
+}
+
+function setAuthorizationHeaderDefault(token) {
+    if(token)
+        axios.defaults.headers.common['Authorization'] = `Bearer ${token}`;
+    else
+        delete axios.defaults.headers.common['Authorization'];
```
Comment (mirzaAlispahic) [inline auction/src/services/SessionService.js]: use curly braces

File: auction/src/services/SessionService.js [hunk 17]
```diff
@@ -0,0 +1,46 @@
...
+    });
+
+    setAuthorizationHeaderDefault(token);
+    Cookies.set('token', token, { expires: 7 });
+}
+
+export const register = (dispatch, registerData) => {
+    axios.post('/register', registerData)
+    .then(res => setUserLoginState(res, dispatch))
+    .catch(err => console.log(err));
```
Comment (mirzaAlispahic) [inline auction/src/services/SessionService.js]: indent

File: auction/src/services/SessionService.js [hunk 18]
```diff
@@ -0,0 +1,46 @@
...
+};
+
+export const login = (dispatch, loginData) => {
+    axios.post('/login', loginData)
+    .then(res => setUserLoginState(res, dispatch))
+    .catch(err => console.log(err));
```
Comment (mirzaAlispahic) [inline auction/src/services/SessionService.js]: indent

**Review by mirzaAlispahic — APPROVED** [review]

**Pull request merged**